clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2022,6 +2035,20 @@ bool CompilerInvocation::ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args,
       Diags.Report(diag::err_drv_invalid_value) << A->getAsString(Args) << Name;
   }
 
+  if (const Arg *A = Args.getLastArg(OPT_mcf_branch_label_scheme_EQ)) {
+    if (Opts.CFProtectionBranch) {
+      const StringRef Scheme = A->getValue();
+      if (Scheme == "unlabeled")
+        Opts.setCFBranchLabelScheme(CFBranchLabelSchemeKind::Unlabeled);
+      else if (Scheme == "func-sig")
+        Opts.setCFBranchLabelScheme(CFBranchLabelSchemeKind::FuncSig);
+      else
+        Diags.Report(diag::err_drv_invalid_value)
+            << A->getAsString(Args) << Scheme;
+    } else
+      Diags.Report(diag::warn_drv_unused_argument) << A->getAsString(Args);
```

Comment:
If you flip the order of the ifs I thin you'll get the unused argument warning automatically. If you don't call `getLastArg` the argument won't be "claimed"

```
if (Opts.CFProtectionBranch)
  if (const Arg *A = Args.getLastArg(OPT_mcf_branch_label_scheme_EQ)) {

  }
}
```

Reply:
Fixed in #109784.